### PR TITLE
sound: Created a Sample Game with Chiventure Structs

### LINKED ISSUE #1377: sound: research and create chiventure structs needed for the sound example
In order to create a demo, we need to research the existing chiventure structs and determine which components would be necessary. After researching these structs, we identified the key structs and created any other structs that would be required. Then, we will create a pull request that runs a demo game before introducing sound, to make sure that we have developed a full understanding of the chiventure structs.

Assignees: Jake, Rhea

### PR DESCRIPTION
**Pull request by sangerr**
In this pull request, we aimed to create a sample game with chiventure structs. This is an initial step before we add in the sound module, in order to familiarize ourselves with the chiventure structs and make sure that we are able to integrate the sound once the sample game can compile. We commented in where the sound structures will go, and we are now moving on to writing those implementations.

Assignees: Jake and Rhea

### CODE REVIEW

**Review by JaStarbuck — APPROVED** [review]
This looks good to me. I'm only concerned about line 38 and 39 since on the debugging hw I didn't do the part right where something was free'd before it was initialized and the only way I got it to run was by removing the free(). However, I did get it wrong so if it works for this I'm all for it.

File: src/sound/examples/sound_example.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    game_t *game = game_new("Welcome to Chiventure!");
+
+    load_normal_mode(game);
+
+    // make room
+    room_t *room1 = room_new("room1", "Room One","Welcome.");
+    add_room_to_game(game, room1);
+    game->curr_room = room1;
+
+    game_free(ctx->game);
```
Comment (ecoble) [inline src/sound/examples/sound_example.c]: Agree with Jason's concerns, why are you freeing the game here?

**Review by ecoble — CHANGES_REQUESTED** [review]
This looks like a good start!

**Review by ecoble — APPROVED** [review]

**Comment by jwschroeder21** [comment]
summary:

In this pull request, we created a sample game with chiventure structs. This is an initial step before we add in the sound modules, in order to familiarize ourselves with the chiventure structs and make sure that we are able to integrate the sound once the sample game can compile. We commented in where the sound structures will go, and we are now moving on to writing those implementations. We have written those implementations, and submitted a PR for them on issue #1320!

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Nice job on this sample game, and great job catching hard to notice issues on the review!

**Pull request merged**